src/librustdoc/clean/mod.rs

```diff
@@ -795,13 +792,6 @@ fn clean_fn_or_proc_macro(
         }
         None => {
             let mut func = clean_function(cx, sig, generics, body_id);
-            let def_id = item.def_id.to_def_id();
-            func.header.constness =
```

Comment:
I noticed several places where the constness is overrided like this. I don't quite understand why this code existed. Does the new, on-demand computation account for the old overriding behavior?

Reply:
Likewise, some places used `is_const_fn_raw` while others used `is_const_fn`. Is this handled properly?

Reply:
It was more or less the same code done by different functions. We have a lot of rustdoc tests for functions and methods to ensure their signature is as expected (I know it because I broke a lot of them while working on this PR).